astropy/stats/funcs.py

```diff
@@ -1019,6 +1019,77 @@ def biweight_midvariance(a, c=9.0, M=None, axis=None):
 
     return (n ** 0.5) * f1 / f2
 
+def biweight_midcovariance(a, c=9.0, M=None):
+    r"""
+    Compute the biweight midcovariance.
+
+    This is a robust and resistant estimator of the covariance matrix.
+
+    For more details, see:
+    `<http://www.itl.nist.gov/div898/software/
+    dataplot/refman2/auxillar/biwmidc.htm>_.`
+
+    Parameters
+    ----------
+    a : array-like, shape=(N_dims, N_samples)
+        Input array
+
+    c : float, optional
+        Tuning constant. Default is 9.0
+
+    M : array-like, optional, shape=(N_dims,)
+        Initial guess for biweight location
+
+    Returns
+    -------
+    biweight_covariance : `~numpy.ndarray`
+        Estimate of the covariance matrix of <a, a.T>
+
+    Examples
+    --------
+    Generate multivariate Gaussian with outliers and compute
+    numpy covariance and the biweight_covariance
+
+    >>> import numpy as np
+    >>> from astropy.stats import biweight_midcovariance
+    >>> rng = np.random.RandomState(1)
+    >>> d = np.array([rng.normal(0,1,200), rng.normal(0,3,200)])
+    >>> d[0,0] = 30.0
+    >>> np_cov = np.cov(d)
+    >>> bw_cov = biweight_midcovariance(d)
+    >>> print(np.around((np.sqrt(np_cov.diagonal()), np.sqrt(bw_cov.diagonal())), 1))
+    [[ 2.3  3.1]
+     [ 0.9  3.1]]
```

Comment:
Not really any particular reason, just something that is an obvious outlier of what you would expect with a normal distribution with scale = 1 or 3. I can add a comment that explains that.